src/data.rs

```diff
@@ -558,6 +558,14 @@ impl<'a> IntoData<'a> for &'a [u8] {
     }
 }
 
+impl<'a> IntoData<'a> for &'a mut [u8] {
+    type Output = Data<'a>;
+
+    fn into_data(self) -> Result<Data<'a>> {
+        Data::from_seekable_reader(Cursor::new(self)).map_err(|e| e.error())
```

Comment:
You probably want `Data::from_seekable_stream` here instead of from_seekable_reader, which will allow writing to the slice.

```suggestion
        Data::from_seekable_stream(Cursor::new(self)).map_err(|e| e.error())
```